### ISSUE TITLE
Pushing compiled PDF to new repo (revisit)

### ISSUE THREAD

**Issue by Siddhant-Ray** [sloorush/resume#4]
TL;DR: Curious to know how you get the part in quotes to work as it didn't work for me (pushing to new repo part)

Hi,

I am curious how you got this part to work, when you push the build PDF to a new repository within the action. Specifically, this part:

> - name: Push to files
        run: |
          git config --global user.email "aarush.bhatt@gmail.com"
          git config --global user.name "Aarush Bhat"
          git clone https://sloorush:${{ secrets.ACCESS_TOKEN }}@github.com/sloorush/sloorush.github.io.git target
          mv resume.pdf ./target
          cd ./target
          git add .
          git diff-index --quiet HEAD || git commit -m "Automatic updation of resume from github.com/sloorush/resume"
          git push origin master
          
The reason I am asking is, I tried to do something very similar, but it kept failing by giving the error:

> fatal: could not read Username for 'https://github.com': No such file or directory

I managed an alternative way (so it works yay) but I am an inherently curious person :P, so I thought I'll ask you if I missed something.

PS: I did set up a Personal Authentication Token and add it as a repository secret for my repo, similar to `secrets.ACCESS_TOKEN ` for you, so I don't think that is an issue. Maybe you carried out an additional step?

PPS: If you want to know about my alternative method, it uses a new action to set up an SSH key-pair authentication and then I  clone via ssh <s>(this is actually quite nice as you can then skip the lines)</s> seems like this is not true, I may have messed up!
 <s>
> git config --global user.email "aarush.bhatt@gmail.com"
          git config --global user.name "Aarush Bhat"
</s>
Why is this nice? Well <s>for one, your personal email is not exposed (more of a privacy thing), plus</s> SSH keys are easier to manage and revoke than authentication tokens.

**Comment by sloorush** [comment]
> `fatal: could not read Username for 'https://github.com': No such file or directory`

I did not face any issues like this. A quick search tells me it is a problem with your git remote ([here](https://stackoverflow.com/questions/22147574/github-fatal-could-not-read-username-for-https-github-com-no-such-file-o)).

It might also be because your Personal Authentication Token lacks the permissions to push or access another repository(just a guess).

Even I thought of setting it up with ssh but this was a simpler and straightforward way and worked well without any issues for me.

#### Alternative methods:
- You can also use the checkout action to clone and push into a repository. I think that is a safer way than using a normal shell script for this ([like this](https://stackoverflow.com/a/72696837/13308517))
- You can use a commit to github action from the github actions store ([here](https://github.com/cpina/github-action-push-to-another-repository)). Not sure how safe this is, I personally haven't seen the code.

**Comment by sloorush** [comment]
> seems like this is not true, I may have messed up

The SSH way should not have this right? What went wrong?

**Comment by Siddhant-Ray** [comment]
> > `fatal: could not read Username for 'https://github.com': No such file or directory`
> 
> I did not face any issues like this. A quick search tells me it is a problem with your git remote ([here](https://stackoverflow.com/questions/22147574/github-fatal-could-not-read-username-for-https-github-com-no-such-file-o)).
> 
> It might also be because your Personal Authentication Token lacks the permissions to push or access another repository(just a guess).
> 
These are the first two things I checked myself, and this wasn't the the issue because it still didn't work. I guess it's not really worth investigating anymore now that it works with SSH.

**Comment by Siddhant-Ray** [comment]
> > seems like this is not true, I may have messed up
> 
> The SSH way should not have this right? What went wrong?

No, it works with SSH, an oversight on my side was that I thought it was working without setting the git config --global user.email but ofc that it is wrong, it cannot work. All good now and closing this.
